tests/test_recorder.js

```diff
@@ -361,128 +373,106 @@ test('rec() throws when reenvoked with already recorder requests', function(t) {
+          t.end();
         });
-        t.end();
-      });
-    }
-  );
-  req.end();
+      }).end('012345');
+  });
 });
 
-// Do not copy tests that rely on the process.env.AIRPLANE, we are deprecating that via #1231
-test('records and replays gzipped nocks correctly', {skip: process.env.AIRPLANE}, function(t) {
+test('records request headers correctly', function(t) {
+  const server = http.createServer((request, response) => response.end());
+  t.once('end', () => server.close());
 
   nock.restore();
   nock.recorder.clear();
   t.equal(nock.recorder.play().length, 0);
 
-  nock.recorder.rec({
-    dont_print: true,
-    output_objects: true
-  });
-
-  var makeRequest = function(callback) {
-    superagent.get('https://bit.ly/1hKHiTe', callback);
-  };
-
-  makeRequest(function(err, resp) {
-
-    t.ok(!err);
-    t.ok(resp);
-    t.ok(resp.headers);
-    t.equal(resp.headers['content-encoding'], 'gzip');
-
-    nock.restore();
-    var nockDefs = nock.recorder.play();
-    nock.recorder.clear();
-    nock.activate();
-
-    // Original bit.ly request is redirected to the target page.
-    t.true(nockDefs.length > 1);
-    var nocks = nock.define(nockDefs);
-
-    makeRequest(function(mockedErr, mockedResp) {
-
-      t.equal(err, mockedErr);
-      t.deepEqual(mockedResp.body, resp.body);
-      t.equal(mockedResp.headers['content-encoding'], 'gzip');
+  server.listen(() => {
+    nock.recorder.rec({
+      dont_print: true,
+      output_objects: true,
+      enable_reqheaders_recording: true
+    });
 
-      _.each(nocks, function(nock) {
-        nock.done();
-      });
+    http.request(
+      {
+        hostname: 'localhost',
+        port: server.address().port,
+        path: '/',
+        method: 'GET',
+        auth: 'foo:bar'
+      },
+      res => {
+        res.resume();
+        res.once('end', function() {
+          nock.restore();
+          const recorded = nock.recorder.play();
+          t.equal(recorded.length, 1);
+          t.type(recorded[0], 'object');
+          t.equivalent(recorded[0].reqheaders, {
+            host: `localhost:${server.address().port}`,
+            'authorization': 'Basic ' + Buffer.from('foo:bar').toString('base64'),
+          });
+          t.end();
+        });
+      }
+    ).end();
+  });
+});
 
-      t.end();
+test('records and replays gzipped nocks correctly', function(t) {
+  const exampleText = '<html><body>example</body></html>';
 
+  const server = http.createServer((request, response) => {
+    zlib.gzip(exampleText, (err, result) => {
+      t.notOk(err)
+      response.writeHead(200, {'content-encoding': 'gzip'});
+      response.end(result);
     });
   });
+  t.once('end', () => server.close());
 
-});
-
-// Do not copy tests that rely on the process.env.AIRPLANE, we are deprecating that via #1231
-test('records and replays gzipped nocks correctly when gzip is returned as a string', {skip: process.env.AIRPLANE}, function(t) {
```

Comment:
It's not clear how this test is supposed to be different from the one before it. Absent a reason to keep it, I removed it.